### PR TITLE
Fix MSI Upgrade failure for preview builds

### LINKED ISSUE #8289: PowerShell-6.2.0-preview.2-win-x64 error on startup (also occurs on upgrade to preview.3)
Steps to reproduce
------------------
pwsh-preview

Expected behavior
-----------------
Launch PowerShell-6.2.0-preview.2

Actual behavior
---------------
Gives the following error:

> Error:
>   An assembly specified in the application dependencies manifest (pwsh.deps.json) was not found:
>     package: 'System.Private.ServiceModel', version: '4.5.3'
>     path: 'runtimes/win/lib/netstandard2.0/System.Private.ServiceModel.dll'

Environment data
----------------

```powershell
> $PSVersionTable

```
N/A

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

- Fix MSI Upgrade failure for preview builds
  - Changing the identifiers every build should resolve this issue.
  - Also, tell MSI to reinstall components if the version changed in any way.

## PR Context

Hopefully really fix #8289 this time

Testing of RC.1 found we still got this error when upgrading from preview.4

```
MSI (s) (4C:A8) [10:18:32:354]: Disallowing installation of component: {A60E946A-2C86-4104-B883-D4B17B50B5B0} since the same component with higher versioned keyfile exists
MSI (s) (4C:A8) [10:18:32:487]: Disallowing installation of component: {7E693303-C675-4985-A1F0-84688E685656} since the same component with higher versioned keyfile exists
MSI (s) (4C:A8) [10:18:32:783]: Disallowing installation of component: {1CCDF75D-482A-4B20-B3D1-2934D374B96B} since the same component with higher versioned keyfile exists
MSI (s) (4C:A8) [10:18:32:787]: Disallowing installation of component: {267520C6-CD61-4FF4-9CA4-83400EAAC3E9} since the same component with higher versioned keyfile exists
MSI (s) (4C:A8) [10:18:32:825]: Disallowing installation of component: {F7B290E8-4DDD-4C20-B3F8-5DD38608DFEF} since the same component with higher versioned keyfile exists
```

With the component also being deleted.

The previous fix didn't change the GUID identifier which triggers the deletion.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@TravisEz13 Could you add more information in the description

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -2726,8 +2718,19 @@ function New-MSIPackage
...
+    # We are verifying that the generated $wixFragmentPath and $FilesWxsPath are functionally the same
     Test-FileWxs -FilesWxsPath $FilesWxsPath -HeatFilesWxsPath $wixFragmentPath
 
+    if ($isPreview)
+    {
+        # Now that we know that the two are functionally the same,
+        # We only need to use $FilesWxsPath for release we want to be able to Path
+        # and two releases shouldn't have the same identifiers,
+        # so we use the generated one for preview
+        $FilesWxsPath = $wixFragmentPath
```
Comment (iSazonov) [inline tools/packaging/packaging.psm1]: Won't this break the update one preview to next preview?

Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: No, this is the code we were using before I added the code to allow for patching.

**Review by adityapatwardhan — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by TravisEz13** [comment]
Merged this so we can cherry-pick into the the RC.1 release

**Pull request merged**